src/CLICommand/cli_command.ts

```diff
@@ -0,0 +1,91 @@
+import { Command } from 'commander';
+import { CliApiObject, ParsedArguments } from './cli';
+import { Parameter, ParameterName } from './parameter';
+
+export type CommandName = string;
+export interface CommandDescriptor {
+	name: CommandName;
+	parameters: ParameterName[];
+	action(options: ParsedArguments): void;
+}
+
+const program: CliApiObject = new Command() as CliApiObject;
+
+// Set up command line option parsing
+//const validActions = ['create-drive', 'rename-drive', 'upload-file'];
+program.option('-h, --help', 'Get help');
+//program.option('create-drive', 'action to create a new drive (and its corresponding root folder)');
+program.addHelpCommand(false);
+
+function setCommanderCommand(commandDescriptor: CommandDescriptor, program: CliApiObject): void {
+	// debugger;
+	let command: CliApiObject = program.command(commandDescriptor.name);
+	commandDescriptor.parameters.forEach((parameterName) => {
+		const parameter = new Parameter(parameterName);
+		const aliasesAsString = parameter.aliases.join(' ');
+		const paramType = (function () {
+			if (parameter.type === 'array') {
+				return `<${parameterName}...>`;
+			} else if (parameter.type === 'boolean') {
+				return '';
+			}
+			return `<${parameterName}>`;
+		})();
+		command = command.option(`${aliasesAsString} ${paramType}`, parameter.description, parameter.default);
+	});
+	command = command.action((options) => {
+		commandDescriptor.action(options);
+	});
+}
+
+export class CLICommand {
+	private static _doneSettingCommands = false;
+	private static _argv?: string[]; // Custom argv vector for testing propuse
+
+	/**
+	 * @param {CommandDescriptor} commandDescription an immputable representation of a command
```

Comment:
spelling nit: "immutable", and below/above on _argv -- "purposes"

Reply:
c0f61a7a2ad689c94578a24172c72cb016135646